### PR TITLE
Use `ProtostarTOML` in `BuildCommand`

### PR DESCRIPTION
**Pull request by kasperski95**
- connected `ProtostarTOML` to `BuildCommand`
- refactored `BuildCommand`

### CODE REVIEW

File: protostar/protostar_toml/protostar_toml_section.py [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 from pathlib import Path
-from typing import Any, Dict, List, Union
+from typing import Any, Dict, Generic, List, TypeVar, Union
 
 from protostar.protostar_toml.protostar_toml_exceptions import (
     InvalidProtostarTOMLException,
 )
 from protostar.utils.protostar_directory import VersionManager, VersionType
 
+T = TypeVar("T")
```
Comment (MaksymilianDemitraszek) [inline protostar/protostar_toml/protostar_toml_section.py:10]:
```suggestion
T = TypeVar("T", bound="ProtostarTOMLSection")
```
Wouldn't be stricter? not sure not an expert in `TypeVar`

**Review by MaksymilianDemitraszek — APPROVED** [review]

File: protostar/commands/build/project_compiler.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+from protostar.commands.build.build_exceptions import CairoCompilationException
+from protostar.protostar_toml.protostar_contracts_section import (
+    ProtostarContractsSection,
+)
+from protostar.protostar_toml.protostar_project_section import ProtostarProjectSection
+from protostar.utils.starknet_compilation import StarknetCompiler
+
+
+class ProjectCompiler:
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply protostar/commands/build/project_compiler.py:19]: Builder suggests that this class implements Builder Pattern.

[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply protostar/commands/build/project_compiler.py:19]: True, Compiler is better

File: protostar/protostar_toml/protostar_project_section.py [hunk 3]
```diff
@@ -11,6 +11,13 @@
 
 @dataclass
 class ProtostarProjectSection(ProtostarTOMLSection):
+    class Loader(ProtostarTOMLSection.Loader):
+        def __init__(self, protostar_toml_reader: ProtostarTOMLReader) -> None:
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply protostar/protostar_toml/protostar_project_section.py:15]: I'd rather keep it annotated for the sake of consistency.

https://mypy.readthedocs.io/en/stable/class_basics.html#annotating-init-methods

**Pull request merged**